### DISCUSSION TITLE
Azure Resource Graph  query requires a subscription

### DISCUSSION THREAD

**Discussion by msbannert** [grafana/grafana#66948] [Data Sources]
When creating any diagram using Azure Monitor data source with an azure resource graph query the UI forces to specify at least one  Subscription to be used. If one leaves the field empty, the application will pick a subscription and fill it in by itself.
However, the query seems to work properly without the subscription field filled.
This aligns with the behaviour in the azure Resource Graph explorer in the azure portal, where you also execute queries always against all available subscriptions (e.g. to all subscriptions to which the current user has access to).
Proposal: get rid of the enforcement of filling in a subscription. If no subscription specified, then the query should run against all accessible subscriptions.

**Comment by kriszentner** [comment]
Seconded. Though the behavior now seems to be that no query will occur if the Subscriptions field is empty. Furthermore, there's no way to query via Management Groups.
I'd propose a check box labeled "all subscriptions" that disables the "subscriptions" box. Though I'd love also to see support for management groups. Some of need to report across hundreds of subscriptions and this functionality would be very useful.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82577. Feel free to continue the discussion around this there. Thank you!
